### PR TITLE
4039 dttp religious education trainees have been incorrectly mapped to philosophy

### PR DESCRIPTION
**Pull request by richardpattinson**
### Context

The ID for philosophy in Dttp::Codesets::CourseSubjects was set to be the same as that of religious education. This means that all religious education courses were being set to philosophy instead. This new service will run through each trainee with a philosophy subject specialism and check the original dttp placement assignment for the true subject specialism and will set the subject accordingly.

### Changes proposed in this pull request

Migration that runs the new service

### Guidance to review

Run the service, check the tests.

### CODE REVIEW

**Comment by richardpattinson** [comment]
[DTTP: Religious education trainees have been incorrectly mapped to philosophy](https://trello.com/c/xg2fz1g9/4039-dttp-religious-education-trainees-have-been-incorrectly-mapped-to-philosophy)

File: app/services/fix_trainees_with_incorrectly_assigned_philosophy_specialisms.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class FixTraineesWithIncorrectlyAssignedPhilosophySpecialisms
+  RELIGIOUS_EDUCATION_DTTP_ID = "82a12838-b3cf-e911-a860-000d3ab1da01"
+  def call
+    trainees = Trainee.where(course_subject_one: CourseSubjects::PHILOSOPHY).or(Trainee.where(course_subject_two: "philosophy")).or(Trainee.where(course_subject_three: "philosophy")).where(created_from_dttp: true)
+    trainees.each do |trainee|
+      if trainee.dttp_trainee.latest_placement_assignment.response["_dfe_ittsubject1id_value"] == RELIGIOUS_EDUCATION_DTTP_ID
```
Comment (edwardhorsford) [inline app/services/fix_trainees_with_incorrectly_assigned_philosophy_specialisms.rb:8]: I think `RELIGIOUS_EDUCATION_DTTP_ID` is misnamed - could it point to the actual code or else `RELIGIOUS_STUDIES_DTTP_ID`?

File: db/data/20220425143919_fix_incorrectly_assigned_philosophy_subject_specialisms.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class FixIncorrectlyAssignedPhilosophySubjectSpecialisms < ActiveRecord::Migration[6.1]
+  def up
+    trainees = Trainee.where(course_subject_one: CourseSubjects::PHILOSOPHY).or(Trainee.where(course_subject_two: CourseSubjects::PHILOSOPHY)).or(Trainee.where(course_subject_three: CourseSubjects::PHILOSOPHY)).where(created_from_dttp: true)
```
Comment (gpeng) [inline db/data/20220425143919_fix_incorrectly_assigned_philosophy_subject_specialisms.rb:5]: Can we split this line?

File: db/data/20220425143919_fix_incorrectly_assigned_philosophy_subject_specialisms.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    trainees.each do |trainee|
+      if trainee.dttp_trainee.latest_placement_assignment.response["_dfe_ittsubject1id_value"] == RELIGIOUS_EDUCATION_DTTP_ID
```
Comment (gpeng) [inline db/data/20220425143919_fix_incorrectly_assigned_philosophy_subject_specialisms.rb:7]: I don't think this is defined `RELIGIOUS_EDUCATION_DTTP_ID`. As per @edwardhorsford 's comment it should reference the codeset RELIGIOUS_STUDIES entity id shouldn't it?

**Review by gpeng — CHANGES_REQUESTED** [review]

**Review by gpeng — APPROVED** [review]

**Pull request merged**